src/jit/gentree.h

```diff
@@ -300,6 +300,153 @@ class FieldSeqStore
     }
 };
 
+class ClassLayout
```

Comment:
I'll confess that I simply dumped this class in the first place where it worked :). I'd be OK with adding new files, though I remember a discussion about the JIT source directory already having lots of files and no subdirectory structure to make it more manageable. Up to the team.

Reply:
Moved all new code to `layout.h`/`layout.cpp`, seems reasonable. Especially if we're going to add more info/logic to `ClassLayout`:
* HFA (and maybe other ABI related info that now gets duplicated in `fgArgTabEntry`) info
* SIMD info
* maybe field info - when promoting we query EE for fields and other info for every promotable variable, even if it has the same type as a previously promoted variable. We could try to add the necessary info to `ClassLayout` the first time we get it.
* maybe it would be even worthwhile to have an abstraction for fields too, instead of using `CORINFO_FIELD_HANDLE` directly